presto-main/src/main/java/com/facebook/presto/sql/planner/QueryPlanner.java

```diff
@@ -390,12 +390,17 @@ private PlanBuilder explicitCoercionFields(PlanBuilder subPlan, Iterable<Express
         projections.putAll(coerce(uncoerced, subPlan, translations));
 
         for (Expression expression : alreadyCoerced) {
-            Symbol symbol = symbolAllocator.newSymbol(expression, analysis.getType(expression));
-            Expression parametersReplaced = ExpressionTreeRewriter.rewriteWith(new ParameterRewriter(analysis.getParameters(), analysis), expression);
-            translations.addIntermediateMapping(expression, parametersReplaced);
-            Expression rewritten = subPlan.rewrite(expression);
-            projections.put(symbol, rewritten);
-            translations.put(parametersReplaced, symbol);
+            if (expression instanceof SymbolReference) {
```

Comment:
I wonder if this branch shouldn't be treated as a special case like in: https://github.com/prestodb/presto/pull/9345/commits/0e47657def7c1f566c34a0c2113e252fe1344aa9#diff-21d31cb9d487772413eff4f67b8555deR339

Reply:
Not sure I understand. It *is* being treated as a special case.

Reply:
My "intuition" is that the main scenario of this method is for generic `expressions`, but there is special case for `SymbolReferences`. Therefore  as in https://github.com/prestodb/presto/commit/0e47657def7c1f566c34a0c2113e252fe1344aa9#diff-21d31cb9d487772413eff4f67b8555deR339, `SymbolReferences` handling is contained in `if` scope ending with `continue`. Generic `expression` handling is not indented. 

But this is just a really minor comment. Feel free to skip it.

Reply:
Ah, I see. You mean do something like this?

```java
for (...) { 
    if (expression instanceof SymbolReferece) {
        ...
        continue;
    }

    ...
}
```

Reply:
yup. Thanks